compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -3090,7 +3090,13 @@ impl<'a> Parser<'a> {
         if let Some((ident, is_raw)) = self.token.lifetime() {
             // Disallow `'fn`, but with a better error message than `expect_lifetime`.
             if matches!(is_raw, IdentIsRaw::No) && ident.without_first_quote().is_reserved() {
-                self.dcx().emit_err(errors::InvalidLabel { span: ident.span, name: ident.name });
+                self.dcx().emit_err(errors::InvalidLabel {
```

Comment:
(alternatively, we could just mimic the diagnostic emitted by `expect_lifetime` ("lifetimes cannot use keyword names") and just say "labels cannot use keyword names"